src/components/PostDetails/PostDetails.tsx

```diff
@@ -1,101 +1,108 @@
-                    </button>
+  const deletecomment = async (id: number) => {
+    if (commentsList) {
+      await delComment(id);
+      findcomments();
+    }
+  };
 
-                    {isHidden && (
-                      <ul className="PostDetails__list">
-                        {postComments.map(comment => (
-                          <li
-                            className="PostDetails__list-item"
-                            key={comment.id}
-                          >
-                            <button
-                              type="button"
-                              className="PostDetails__remove-button button"
-                              onClick={() => {
-                                removePostComment(comment.id);
-                                setpPostComments(
-                                  postComments.filter(c => c.id !== comment.id
-                                  ));
-                              }}
-                            >
-                              X
-                            </button>
-                            <p>{comment.body}</p>
-                          </li>
-                        ))}
-                      </ul>
-                    )}
-                  </>
-                )}
-            </section>
+  useEffect(() => {
+    findcomments();
+  }, [postId]);
 
-            <section>
-              <div className="PostDetails__form-wrapper">
-                <NewCommentForm
-                  postId={postId}
-                  fetchComments={fetchComments}
-                />
-              </div>
-            </section>
-          </>
-        )}
+  return (
+    <div className="PostDetails">
+      <h2>Post details:</h2>
+      {postId !== 0 && (
+        <>
+          <section className="PostDetails__post">
+            <p>
+              {postDetails?.title}
+            </p>
+          </section>
+          <section className="PostDetails__comments">
+            <button
+              type="button"
+              className="button"
+              onClick={() => {
+                setVisiblecomments(!visiblecomments);
+              }}
+            >
+              {commentsList && ((visiblecomments)
+                ? `Show ${commentsList.length} comments`
+                : `Hide ${commentsList.length} comments`)}
+            </button>
+            <ul
+              className={visiblecomments
+                ? 'PostDetails__visiblelist'
+                : 'PostDetails__list'}
+            >
+              {commentsList && commentsList.map(comm => (
+                <li
+                  className="PostDetails__list-item"
+                  key={comm.id}
+                >
+                  <button
+                    type="button"
+                    className="PostDetails__remove-button button"
+                    onClick={() => {
+                      deletecomment(comm.id);
+                    }}
```

Comment:
```suggestion
                    onClick={() => deletecomment(comm.id)}
```
curly brackets are optional